crates/cli/tests/dylib_test.rs

```diff
@@ -36,6 +36,36 @@ fn test_dylib() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn test_dylib_with_error() -> Result<()> {
```

Comment:
This new test function shares a lot of code with the function above, can it be deduplicated?